pl_examples/hydra_examples/pl_hydra/trainer_conf.py

```diff
@@ -0,0 +1,257 @@
+from dataclasses import dataclass, asdict
+from typing import Optional, List, Dict, Any
+
+from hydra.utils import instantiate
+from hydra.core.config_store import ConfigStore
+from hydra.types import ObjectConf
+
+
+cs = ConfigStore.instance()
+
+
+@dataclass
+class LightningTrainerConf:
+    # callbacks: Optional[List[Callback]] = None
+    default_root_dir: Optional[str] = None
+    gradient_clip_val: float = 0
+    process_position: int = 0
+    num_nodes: int = 1
+    num_processes: int = 1
+    gpus: Optional[Any] = None  # todo
+    auto_select_gpus: bool = False
+    tpu_cores: Optional[Any] = None  # todo
+    log_gpu_memory: Optional[str] = None
+    progress_bar_refresh_rate: int = 1
+    overfit_batches: float = 0.0  # todo
+    track_grad_norm: Any = -1  # todo
+    check_val_every_n_epoch: int = 1
+    fast_dev_run: bool = False
+    accumulate_grad_batches: Any = 1  # todo
+    max_epochs: int = 1000
+    min_epochs: int = 1
+    max_steps: Optional[int] = None
+    min_steps: Optional[int] = None
+    limit_train_batches: float = 1.0  # todo
+    limit_val_batches: float = 1.0  # todo
+    limit_test_batches: float = 1.0  # todo
+    val_check_interval: float = 1.0  # todo
+    log_save_interval: int = 100
+    row_log_interval: int = 50
+    distributed_backend: Optional[str] = None
+    precision: int = 32
+    print_nan_grads: bool = False  # backward compatible todo: remove in v0.9.0
+    weights_summary: Optional[str] = "top"
+    weights_save_path: Optional[str] = None
+    num_sanity_val_steps: int = 2
+    truncated_bptt_steps: Optional[int] = None
+    resume_from_checkpoint: Optional[str] = None
+    benchmark: bool = False
+    deterministic: bool = False
+    reload_dataloaders_every_epoch: bool = False
+    auto_lr_find: Any = False
+    replace_sampler_ddp: bool = True
+    terminate_on_nan: bool = False
+    auto_scale_batch_size: Any = False
+    prepare_data_per_node: bool = True
+    amp_level: str = "O1"
+    num_tpu_cores: Optional[int] = None
+
+
+cs.store(group="pl/trainer", name="trainer", node=LightningTrainerConf)
+
+
+@dataclass
+class ModelCheckpointConf:
+    filepath: Optional[str] = None
+    monitor: str = "val_loss"
+    verbose: bool = False
+    save_last: bool = False
+    save_top_k: int = 1
+    save_weights_only: bool = False
+    mode: str = "auto"
```

Comment:
potential for using an Enum.
This might need a deeper change in PL so maybe not.

Reply:
Agreed, this is where it is currently defined: https://github.com/PyTorchLightning/pytorch-lightning/blob/66ffbaddf5abf3e3fca5789ba2427be55b27d1af/pytorch_lightning/callbacks/early_stopping.py#L53